File: src/main/java/guru/springframework/spring6webapp/controller/BookControllerThymeleaf.java
package guru.springframework.spring6webapp.controller;

import guru.springframework.spring6webapp.service.BookService;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@AllArgsConstructor
public class BookControllerThymeleaf {
    private BookService bookService;

    @RequestMapping("/books/html")
    public String findAll(Model model) {
        model.addAttribute("books", bookService.findAll());
        return "books";
    }
}

File: src/main/kotlin/guru/springframework/spring6webapp/controller/BookControllerThymeleaf.kt
package guru.springframework.spring6webapp.controller

import guru.springframework.spring6webapp.service.BookService
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.RequestMapping

@Controller
class BookControllerThymeleaf(private val bookService: BookService) {
    @RequestMapping("/books/html")
    fun findAll(model: Model): String {
        model.addAttribute("books", bookService.findAll())
        return "books"
    }
}
